lib/kubeclient/watch_stream.rb

```diff
@@ -3,6 +3,7 @@
 module Kubeclient
   module Common
     # HTTP Stream used to watch changes on entities
+    # Note: this IS NOT(!) a supported interface, and might change without a version bump
```

Comment:
I think only `WatchStream.new` is internal here.  This *is* an object users get back from `.watch_pods` etc., and `.watch` and `.finish` are methods users interact with, both mentioned in README I think.